lib/sprinkler/devices/watering_policy.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Sprinkler.Devices.WateringPolicy do
+  def should_irrigate(moisture_input, temperature_input, humidity_input) do
+    unless soil_moisture_wet?(moisture_input) do
+      when_soil_moisture_not_wet(moisture_input, temperature_input, humidity_input)
+    end
+  end
+
+  defp when_soil_moisture_not_wet(moisture_input, temperature_input, humidity_input) do
+    if soil_moisture_medium?(moisture_input) do
+      when_soil_moisture_medium(temperature_input, humidity_input)
+    else
+      when_soil_moisture_not_medium(temperature_input, humidity_input)
+    end
+  end
+
+  defp when_soil_moisture_medium(temperature_input, humidity_input) do
+    if ambience_temperature_high?(temperature_input) do
+      irrigate('medium')
+    else
+      unless humidity_high?(humidity_input) do
+        irrigate('low')
+      end
+    end
+  end
+
+  defp when_soil_moisture_not_medium(temperature_input, humidity_input) do
+    if humidity_high?(humidity_input) do
+      if ambience_temperature_high?(temperature_input) do
+        irrigate('high')
+      else
+        irrigate('medium')
+      end
+    else
+      irrigate('high')
+    end
+  end
+
+  defp soil_moisture_wet?(moisture_input) do
+    # TODO: check this values, 70%
+    if moisture_input > 70 do
```

Comment:
Maybe we can use environment variables for these values, wdyt?

Reply:
Yes, it was on my mind, but have to investigate how I can do it 😄